operation/fhir-operation-bulkdata/src/main/java/org/linuxforhealth/fhir/operation/bulkdata/model/transformer/JobIdEncodingTransformer.java

```diff
@@ -50,86 +40,54 @@ public static JobIdEncodingTransformer getInstance() {
         return transformer;
     }
 
-    /*
-     * gets the tenant specific encryption key.
-     * @return
+    /**
+     * Create the tenant-specific Hashids object with a "salt" from fhir-server-config
+     * @return the Hashids object or null if no salt is configured in fhir-server-config
      */
-    private SecretKeySpec getJobIdEncryptionKey() {
-        String encryptionKey = ConfigurationFactory.getInstance().getCoreBatchIdEncryptionKey();
-        SecretKeySpec secretKey = null;
-
-        if (encryptionKey != null && !encryptionKey.isEmpty()) {
-            try {
-                byte[] keyBytes = encryptionKey.getBytes("UTF-8");
-                keyBytes = Arrays.copyOf(MessageDigest.getInstance("SHA-1").digest(keyBytes), 16);
-                secretKey = new SecretKeySpec(keyBytes, "AES");
-            } catch (Exception e) {
-                logger.log(Level.WARNING, "Fail to generate encryption key from config!", e);
-            }
-        }
+    private Hashids createHashids() {
+        String salt = ConfigurationFactory.getInstance().getCoreBatchIdEncodingKey();
 
-        if (secretKey == null) {
-            logger.warning("Failed to get encryption key, JavaBatch Job ids will not be encrypted!");
+        if (salt != null && !salt.isEmpty()) {
+            return new Hashids(salt, MIN_ENCODED_LENGTH);
+        } else {
+            return null;
```

Comment:
Should this be an exception?

Reply:
currently we allow the `coreBatchIdEncryptionKey` to be omitted in which case we use the numeric job id.  i'm not sure if thats really necessary, but it does make testing slightly nicer (because the job ids are numeric instead of a cryptic string).

basically if this comes back null then there is no encoding/decoding (but not an error)

Reply:
OK, I think it's fine at this time - no need to make the change bigger. I'm just not a fan of that (original) design because it doesn't really adhere to separation of concerns. My personal choice would be to skip the encode/decode call if not required by the config. It's a subtle point, but it means that if we were to use a different encoder/decoder implementation, that too would need to handle null keys...but this is being a bit purist. Still, always interesting to think about the ramifications of a particular API design.